### PR TITLE
fix(#86): タイポしたスラッシュコマンドで Bot がアイドル化する問題を slash-prefix strip で防止

### LINKED ISSUE #86: fix: タイポしたカスタムコマンド入力で Bot がアイドル状態に陥る (Discord 経由)
## 概要

Discord セッション内でカスタムコマンド名をタイポすると、Bot がアイドル状態に陥り後続のメッセージに反応しなくなる。Bot を再度メンションするまで復旧しない。

直近の Supervisor 改修（PR #84 専用 tmux socket、PR #77 copy-mode 救済）でも本症状は残存している模様。

## 再現手順

1. Discord の任意のセッションチャンネルで Bot をメンションし、**存在しないコマンド名**（タイポ）を投稿:

    ```
    /hanle-review XXX
    ```

    （正しくは `/handle-reviews`）

Image

    別パターン:

    ```
     /sesssion-resume
    ```

    （正しくは `/resume-session`）

Image

2. 観察:

    ```
    Bot はコマンドを受理した形跡を見せず、以降のメンションにも応答しない（アイドル状態）
    Expected: 「unknown command」等のエラーメッセージを返す or 通常の自然言語入力として処理される
    Actual: 無反応のまま、ユーザーがもう一度メンションすると復旧することがある
    ```

## 統合ジャーニーAC

1. **操作**: Discord セッションチャンネルで `/hanle-review` (タイポ) を投稿
   - **期待結果**: Bot が即座に「unknown command: /hanle-review」等のエラー応答を返す、または `/` で始まる文字列をそのまま自然言語入力として Claude Code に転送する
   - **検証手段**: Discord チャンネルで Bot からの応答メッセージが 30 秒以内に届く（メッセージ ID を fetch_messages で確認）
2. **操作**: 同セッションで続けて `@Bot こんにちは` のような通常メッセージを投稿
   - **期待結果**: Bot が通常通り応答する（アイドル状態に陥っていない）
   - **検証手段**: Discord チャンネルで Bot からの応答が 30 秒以内に届く
3. **操作**: `/sesssion-resume` (別パターンのタイポ) でも 1〜2 を再実施
   - **期待結果**: 同じく正常応答
   - **検証手段**: 同上

## コンポーネントAC

- AC-1: Supervisor / claude-hub 側で「未知の slash command を受信したときの fallback 処理」が定義されている（コード上で確認可能）
- AC-2: 上記再現手順を 3 回連続で実施しても Bot がアイドル状態に陥らない（手動 E2E）
- AC-3: 関連ユニットテスト（不明なコマンド受信時の挙動）が追加され green

**完了条件**: 上記再現手順で本バグが**再現しない**こと。

## 環境

- Branch: detached HEAD（直近 #83 #84 merged 後）
- Recent commits: 3fe2c01 fix(#83) Supervisor 専用 tmux socket、be3f205 PR #77 copy-mode 救済
- 観測チャンネル: `1497489528951148615`
- 報告者: takoyaki0778 (2026-04-25)

## 関連

- #72 epic: Discord ↔ Claude Code 通信詰まり問題の再発防止
- #79 Epic: tmux send-keys silent drop 根本原因特定
- メモリ: `feedback_claudehubexit_scope.md`（requireMention=true 仕様）

### PR DESCRIPTION
**Pull request by miyashita337**
## 概要

Discord スレッドで `/hanle-review`（`/handle-reviews` のタイポ）など typo の slash command を投稿すると、Claude Code の Ink TUI がスラッシュコマンドピッカーに入って閉じる入力が来ないまま固まり、`enqueueForThread` の per-thread relay queue が `RELAY_TIMEOUT_MS` (5分) まで block される。これが「Bot がアイドル化」の正体だった。

修正: `MessageCreate` handler で `^/<word>(?:\s|$)` パターン（path 系を除く）を検知した場合、先頭の `/` を除去して自然言語として Claude Code に渡す。

Closes #86

## 設計判断

| 案 | Pros | Cons | 採用 |
|---|---|---|---|
| (a) `/<word>` を検知してエラー応答だけ返し relay しない | bug 完全防止 | `/help` 等の正当な Claude Code slash command も殺す | ❌ |
| (b) `/` を strip して自然言語として forward | 正当 command も自然言語として通る | TUI 経由の slash command 機能が Discord から呼べない（元から呼べていなかった） | ✅ |
| (c) Escape × N + send-keys で picker を抜ける | 既存仕様維持 | picker 抜け方が UI モードで挙動異なる、不安定 | ❌ |

→ 統合ジャーニーAC「`/` で始まる文字列をそのまま自然言語入力として Claude Code に転送する」を満たすため (b) を採用。

`/usr/bin/ls` や `/Users/x/foo` は narrow regex で **path として除外**（先頭 token の後が `/` で続くため `(?:\s|$)` に引っかからない）。

## 変更ファイル (3)

| ファイル | 内容 |
|---|---|
| `supervisor/src/session/slash-prefix.ts` (新規) | `looksLikeSlashCommand()` / `stripLeadingSlash()` の pure utility |
| `supervisor/src/bot.ts` | MessageCreate で `looksLikeSlashCommand` チェック、検知時は thread に警告送信 + stripped を relay |
| `supervisor/tests/session/slash-prefix.test.ts` (新規) | 26 テスト（slash-command shape / path / 自然言語 / edge cases） |

### コンポーネントAC
| AC | 内容 | コマンド | 結果 | 判定 |
|---|---|---|---|---|
| AC-1 | 未知 slash command の fallback 処理がコード上で確認可能 | `grep stripLeadingSlash supervisor/src/bot.ts` | hit | ✅ PASS |
| AC-2 | 再現手順を 3 回連続で Bot がアイドル化しない | (実機 E2E) | merge 後 Supervisor 再起動で確認 | ⏭️ 手動 |
| AC-3 | ユニットテスト追加 + green | `bun test supervisor/tests/session/slash-prefix.test.ts` | 26 pass / 0 fail | ✅ PASS |

加えて `bun build supervisor/index.ts` で 524 modules ビルド成功。

### 統合ジャーニーAC
| 操作 | 期待結果 | 検証手段 |
|---|---|---|
| Discord で `/hanle-review` 投稿 | Bot が即座に「`/` 除去して送信」warning + Claude が自然言語 `hanle-review` として処理 | Discord 上の応答メッセージ確認 |
| 続けて `@Bot こんにちは` | 通常応答（queue は block されない） | 30 秒以内応答 |
| `/sesssion-resume` 別 typo | 同上 | 同上 |

→ 実機 E2E は merge 後 Supervisor 再起動 + Discord 操作で確認。

## 影響範囲

- **後方互換性**: あり。`/help` 等の Claude Code slash command を Discord メッセージ経由で**呼べていた場合**、それは natural language として処理されるようになる。ただし元から TUI picker は典型的に hang していたので実質的な機能差はない。
- **Discord-side `/session` slash command**: 影響なし。`Events.InteractionCreate` で別途処理されるため `MessageCreate` には来ない。

## 関連

- 起点 Issue: #86
- 親 Epic: #72 (silent fail / 通信詰まり)
- 関連 fix: PR #84 (tmux socket), PR #77 (copy-mode)

## リリースノート

* **新機能**
  * ボットがスラッシュコマンド形式のメッセージを自動検出し、先頭のスラッシュを削除するようになりました
  * コマンド形式が検出された場合、スレッドに日本語の情報通知を自動投稿します
  * メッセージ処理前の前処理機能により、より正確なメッセージフォワーディングが実現されました

### CODE REVIEW

File: supervisor/src/bot.ts [hunk 1]
```diff
@@ -196,6 +200,27 @@ export async function startBot(token: string): Promise<void> {
...
+      console.log(
+        `[Bot] Stripped slash prefix in thread ${threadId}: "${original.slice(0, 80)}" → "${messageText.slice(0, 80)}"`
+      );
+      try {
+        await thread.send(
+          `ℹ️ \`/\` 始まりの入力は Claude Code TUI のスラッシュピッカーで詰まる現象を避けるため \`/\` を除去して送信します（自然言語として処理されます。Issue #86）。`
+        );
+      } catch {
+        // Don't let the heads-up notification block the actual relay
+      }
```
[the remark this answers is not in the document]
Comment (miyashita337) [reply supervisor/src/bot.ts]: 対応しました（93b2777）。

提案通り `thread.send().catch(...)` の fire-and-forget に変更し、enqueueForThread 投入前で await しないようにしました。これで Discord API レイテンシ中に同一スレッドの次メッセージが先に queue へ到達する race を防げます。.catch() では `console.warn` で送信失敗を log に残し silent fail を回避しています。

bun build → 524 modules success / bun test slash-prefix → 26 pass.

**Comment by miyashita337** [comment]
@coderabbitai PR-level nitpick 3 件すべて採用しました（0872255）。

| # | 指摘 | 対応 |
|---|---|---|
| 1 | `slash-prefix.ts`: `text.slice(1)` 推奨 | `looksLikeSlashCommand` で先頭 `/` 確定済みのため `text.replace(/^\//, "")` → `text.slice(1)` に変更。コメントで意図を明記 |
| 2 | `slash-prefix.test.ts`: `\s` 改行/タブ網羅追加 | `looksLikeSlashCommand` の `test.each` に `["/help\n続き", true]`, `["/help\targ", true]` 追加 + `stripLeadingSlash` に `"/help\n続き"` → `"help\n続き"`, `"/help\targ"` → `"help\targ"` の 2 ケース追加 |
| 3 | `bot.ts`: PII 配慮で本文 log を抑制 | `original.slice(0, 80)` の full-preview 出力を「token (先頭 32 char) + len」形式に変更。L246 の他 relay log と粒度を統一 |

bun test supervisor/tests/session/slash-prefix.test.ts → 30 pass / 0 fail（既存 26 + 追加 4）。
